Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/qrcode/DccTicketingQrCodeHandler.kt

```diff
@@ -34,6 +47,12 @@ class DccTicketingQrCodeHandler @Inject constructor(
         )
     }
 
+    private fun Set<DccTicketingServiceProviderAllowListEntry>.validateServiceIdentity(serviceIdentity: String) {
+        if (!qrCodeSettings.checkServiceIdentity.value) return
+        val hash = serviceIdentity.toSHA256().encodeUtf8()
```

Comment:
`enodeUtf8()` isn't working as expected , please use `serviceIdentity.encode().sha256()`?
```suggestion
        val hash = serviceIdentity.encode().sha256()
```

Currently  the comparison is failing 
```diff
- serviceIdentityHash=[text=8646c015ffec92cfec73220f602c4892267c054f0eca4aff4d1b40238d394561]
+ ServiceProviderAllowList=[DccTicketingServiceProviderAllowListEntry(serviceIdentityHash=[hex=8646c015ffec92cfec73220f602c4892267c054f0eca4aff4d1b40238d394561])]
find=null
```
hashes are the same ,but the key is different `text=..` ,`hash=..`